### ISSUE TITLE
Error loading vision model in 0.51

### ISSUE THREAD

**Issue by elephantpanda** [microsoft/onnxruntime-genai#1073]
c#  0.51 Directml

`OnnxRuntimeGenAIException: Error encountered while parsing 'D:\Phi3OnnxVision\genai_config.json' JSON Error: Unknown value: visual_features at line 48 index 53`

Didn't have this error when running in 0.40

**Comment by kunal-vaishnavi** [comment]
New ONNX models for [Phi-3 vision](https://huggingface.co/microsoft/Phi-3-vision-128k-instruct-onnx) and [Phi-3.5 vision](https://huggingface.co/microsoft/Phi-3.5-vision-instruct-onnx) have been uploaded to support no-image, single-image, and multi-image scenarios. You can download the new ONNX models to use with ONNX Runtime GenAI v0.5.1.

**Comment by elephantpanda** [comment]
Downloaded the new model. 
After it loads I get the error:
`OnnxRuntimeGenAIException: [ImageDecoder]: Failed when CoInitialize.`
when it does this line:
**processor = new MultiModalProcessor(model);**

Previously there were onnx models specifically for DML or CUDA. Not sure if that is a reason.

**Comment by kunal-vaishnavi** [comment]
> ```
> OnnxRuntimeGenAIException: [ImageDecoder]: Failed when CoInitialize.
> Microsoft.ML.OnnxRuntimeGenAI.Result.VerifySuccess (System.IntPtr nativeResult) (at D:/a/_work/1/onnxruntime-genai/src/csharp/Result.cs:25)
> Microsoft.ML.OnnxRuntimeGenAI.MultiModalProcessor..ctor (Microsoft.ML.OnnxRuntimeGenAI.Model model) (at D:/a/_work/1/onnxruntime-genai/src/csharp/MultiModalProcessor.cs:15)
> Main.Start () (at Assets/Main.cs:37)
> ```

Can you try running the [`phi3v.py`](https://github.com/microsoft/onnxruntime-genai/blob/main/examples/python/phi3v.py) example script to verify the models can run with Python? This will help narrow down whether the problem is specific to the C# example or not.

> Previously there were onnx models specifically for DML or CUDA. Not sure if that is a reason.

The uploaded ONNX model in the GPU folder should work for both CUDA and DML. We're working on adding the support needed to run the same ONNX model on CPU, CUDA, and DML.

**Comment by skyline75489** [comment]
@elephantpanda What version of Windows are you using? We currently rely on [WIC](https://learn.microsoft.com/en-us/windows/win32/wic/-wic-about-windows-imaging-codec) for image decoding. The failure seems to be related to initializing WIC.

**Comment by elephantpanda** [comment]
> [@elephantpanda](https://github.com/elephantpanda) What version of Windows are you using? We currently rely on [WIC](https://learn.microsoft.com/en-us/windows/win32/wic/-wic-about-windows-imaging-codec) for image decoding. The failure seems to be related to initializing WIC.

I am using a Windows 10 cloud computer to run this. So I don't really have control over upgrading this to Windows 11.

If I am to release software to the public which runs GenAI I would expect it to work on Windows 10 and Windows 11. If it is relying on Windows 11 only libraries that doesn't seem great. Unless I can add those libraries to my software.

**Comment by skyline75489** [comment]
@elephantpanda Yes, the WIC dependency was newly introduced in 0.5.0. And since it works with Vista+, Windows 10 shouldn't be a problem. Could you please check if these [libraries](https://learn.microsoft.com/en-us/windows/win32/wic/-wic-api#library-files) are available in your environment? Especially `windowscodecs.dll` which should be loaded at runtime.

**Comment by skyline75489** [comment]
Hi @elephantpanda are you still having the `Failed when CoInitialize` issue?

**Comment by skyline75489** [comment]
This issue has no activity in the last month. I'll close it for now. 0.6.0 is soon to be released and please try the new version to see if helps. Feel free to continue the discussion.

**Comment by dhhunter** [comment]
Greetings everyone!  I just attempted to load up the Vision model using 0.6.0-rc1 and I am sadly also experiencing this issue!

[ImageDecoder]: Failed when CoInitialize. HRESULT: -2147417850

**Comment by dhhunter** [comment]
@skyline75489 It seems that my problem has something to do with Avalonia!  I cloned your repo and was able to run your C# sample Console application just fine!  I created a new Avalonia application out of curiosity and it also blew up with with the exception:

[ImageDecoder]: Failed when CoInitialize. HRESULT: -2147417850.  

I can confirm that I have WIC installed on my system because the Console application has access to it!  I have no idea why it does not like Avalonia though!  I could not find any code in the sample application that setup WIC and attempting to add windowscodecs.dll as a project reference to my application fails.  Do you have any ideas?  Thanks and GOD Bless!  FYI: I am currently using 0.6.0 cpu!

**Comment by skyline75489** [comment]
Hi @dhhunter The error is because GenAI tries to do `CoInitialize` but with different parameter than Avalonia itself (See [link about RPC_E_CHANGED_MODE (0x80010106)](https://learn.microsoft.com/en-us/troubleshoot/windows/win32/coinitializeex-fails-after-calling-htmlhelp-same-thread)). This is tricky but I think there is a better way of handling this.

@wenbingl This is going to be a problem for people tries to use ortx in such scenarios.

**Comment by dhhunter** [comment]
@skyline75489 Thank you very much!  Wrapping it in a background thread worked just fine (assuming I don't use an OgaHandle anyway, whatever that is?)!  Running it Async was going to be a necessary step to making the UI functional anyway!  So it is of minor inconvenience to have to wrap it in a Task, I suppose it is more that it is really not intuitive from a new users point of view!  Thanks again, I am excited to finally get to play with Phi-Vision!

**Comment by ShyRed** [comment]
@skyline75489 @dhhunter Thank you so much for figuring this out.

I had the same issue: Avalonia UI + Vision Model = [ImageDecoder]: Failed when CoInitialize. HRESULT: -2147417850.

In my project I'm using Microsoft.ML.OnnxRuntimeGenAI" Version="0.10.0" and "Microsoft.ML.OnnxRuntimeGenAI.Managed" Version="0.10.1", so the problem with WIC is still persisting.

The background thread workaround works like a charm, but it is quite time consuming to find this issue here and the solution. Would it be possible to put in some sort of disclaimer or a more elaborate exception message?
